client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java

```diff
@@ -254,20 +368,39 @@ private static Long getMaxValue(Field f) throws Exception {
      * @param f the reflection field that is annotated with the Min annotation.
      * @return the Long with the specific minimum value (as a literal) if the annotation is present, otherwise returns null.
      */
-    private static Long getMinValue(Field f) throws Exception {
-        return getLongElement(f, "javax.validation.constraints.Min", "value");
+    private static Long getMinValue(Field f, NameSpace namespace) throws Exception {
+        Objects.requireNonNull(namespace);
+        final String minAnnotationName = getAnnotationName(namespace, JAVAX_MIN_ANNOTATION_NAME, JAKARTA_MIN_ANNOTATION_NAME);
+        return getLongElement(f, minAnnotationName, "value");
+    }
+
+    private static String getDecimalMinValue(Field f, NameSpace namespace) throws Exception {
+        Objects.requireNonNull(namespace);
+        final String decimalMinAnnotationName = getAnnotationName(namespace, JAVAX_DECIMAL_MIN_ANNOTATION_NAME, JAKARTA_DECIMAL_MIN_ANNOTATION_NAME);
+        return getStringElement(f, decimalMinAnnotationName, "value");
     }
 
-    private static String getDecimalMinValue(Field f) throws Exception {
-        return getStringElement(f, "javax.validation.constraints.DecimalMin", "value");
+    private static String getDecimalMaxValue(Field f, NameSpace namespace) throws Exception {
+        Objects.requireNonNull(namespace);
+        final String decimalMaxAnnotationName = getAnnotationName(namespace, JAVAX_DECIMAL_MAX_ANNOTATION_NAME, JAKARTA_DECIMAL_MAX_ANNOTATION_NAME);
+        return getStringElement(f, decimalMaxAnnotationName, "value");
     }
 
-    private static String getDecimalMaxValue(Field f) throws Exception {
-        return getStringElement(f, "javax.validation.constraints.DecimalMax", "value");
+    private static String getPatterRegExp(Field f, NameSpace namespace) throws Exception {
+        Objects.requireNonNull(namespace);
+        final String patternAnnotationName = getAnnotationName(namespace, JAVAX_PATTERN_ANNOTATION_NAME, JAKARTA_PATTERN_ANNOTATION_NAME);
+        return getStringElement(f, patternAnnotationName, "regexp");
     }
 
-    private static String getPatterRegExp(Field f) throws Exception {
-        return getStringElement(f, "javax.validation.constraints.Pattern", "regexp");
+    private static String getAnnotationName(NameSpace namespace, String javaxAnnotationName, String jakartaAnnotationName) {
+        switch (namespace) {
+            case JAVAX:
+                return javaxAnnotationName;
```

Comment:
need "assert" here (as it is a private method, and not a public one). assert that `javaxAnnotationName` starts with `javax.`, and other way round for `jakartaAnnotationName`

Reply:
ack